genomepy/caching.py

```diff
@@ -1,26 +1,87 @@
 
 def clean():
     """Remove cached data on providers."""
     rmtree(genomepy_cache_dir, ignore_errors=True)
     os.makedirs(genomepy_cache_dir, exist_ok=True)
     print("All clean!")
+
+def hybcache(exp_short=3600, exp_long=3600, cache=disk_cache, lock=Lock()):
+    def decorator(func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            # Generate the cache key from the function's arguments.
+            print(args[0].__dict__)
+            key = [func.__name__] + list(args) + list(sorted(kwargs.items()))
+            #key = hash(key)
+            # Pickle object
+            ext = ".joblib"
+            pickle_obj = os.path.join(disk_cache.directory,f"{key}{ext}") #Check if value exists in the short term cache
+            value = cache.get(key)
+            #Check if the function call deos not exists in the memory cache
+            if value is None:
+                #Check if the value exists as pickle and if it is expired
+                with lock:
+                    if os.path.exists(pickle_obj):
+                        current_time = time.time()
+                        expiry_time = os.path.getctime(pickle_obj) + exp_long
+                        # Remove pickled object (long-term) if expired 
+                        if (current_time > expiry_time):
+                            print(f"Removing {key} from short-term memory since it is expired")
+                            os.remove(pickle_obj)
+                            # We need to re-compute and store the result in memory and as pickle
+                            print(f"re-computing {key} and caching value")
+                            value = func(*args, **kwargs)
+                            print(value)
+                            cache.close()
+                            with Cache(cache.directory) as reference:
+                                reference.set(key, value, exp_short)
+                            # Store function result as pickle
+                            dump(value, os.path.join(disk_cache.directory,pickle_obj), protocol=3, compress=True)
+                        else:
+                            # Load existing pickle into memory and skip computation
+                            print(f"Loading {key} from pickle into cache")
+                            with open(pickle_obj, 'rb') as pickle_handle:
+                                cache.close() 
+                                with Cache(cache.directory) as reference:
+                                    reference.set(key, load(pickle_obj), exp_short)
+                            value = cache.get(key)
+                    else:
+                        # Run the function and cache the result for next time.
+                        print(f"computing {key} and storing value")
+                        value = func(*args, **kwargs)
+                        print(value)
+                        cache.close()
+                        with Cache(cache.directory) as reference:
```

Comment:
in line 23 you state: `disk_cache = Cache(directory=genomepy_cache_dir)`
should you not write `with disk_cache.directory as reference:` or something similar?